server/datastore/mysql/setup_experience_test.go

```diff
@@ -37,6 +38,274 @@ func TestSetupExperience(t *testing.T) {
+		InstallScript:     "banana",
+		PreInstallQuery:   "SELECT 3",
+		PostInstallScript: "apple",
+		InstallerFile:     bytes.NewReader([]byte("hello")),
+		StorageID:         "storage3",
+		Filename:          "file3",
+		Title:             "Software2",
+		Version:           "3.0",
+		Source:            "apps",
+		SelfService:       true,
+		UserID:            user1.ID,
+		TeamID:            &team2.ID,
+		Platform:          string(fleet.MacOSPlatform),
+	})
+	require.NoError(t, err)
+
+	ExecAdhocSQL(t, ds, func(q sqlx.ExtContext) error {
+		_, err := q.ExecContext(ctx, "UPDATE software_installers SET install_during_setup = 1 WHERE id IN (?, ?)", installerID1, installerID2)
+		return err
+	})
+
+	app1 := &fleet.VPPApp{Name: "vpp_app_1", VPPAppTeam: fleet.VPPAppTeam{VPPAppID: fleet.VPPAppID{AdamID: "1", Platform: fleet.MacOSPlatform}}, BundleIdentifier: "b1"}
+	vpp1, err := ds.InsertVPPAppWithTeam(ctx, app1, &team1.ID)
+	require.NoError(t, err)
+
+	app2 := &fleet.VPPApp{Name: "vpp_app_2", VPPAppTeam: fleet.VPPAppTeam{VPPAppID: fleet.VPPAppID{AdamID: "2", Platform: fleet.MacOSPlatform}}, BundleIdentifier: "b2"}
+	vpp2, err := ds.InsertVPPAppWithTeam(ctx, app2, &team2.ID)
+	require.NoError(t, err)
+
+	ExecAdhocSQL(t, ds, func(q sqlx.ExtContext) error {
+		_, err := q.ExecContext(ctx, "UPDATE vpp_apps_teams SET install_during_setup = 1 WHERE adam_id IN (?, ?)", vpp1.AdamID, vpp2.AdamID)
+		return err
+	})
+
+	var script1ID, script2ID int64
+	ExecAdhocSQL(t, ds, func(q sqlx.ExtContext) error {
+		res, err := insertScriptContents(ctx, q, "SCRIPT 1")
+		if err != nil {
+			return err
+		}
+		id1, _ := res.LastInsertId()
+		res, err = insertScriptContents(ctx, q, "SCRIPT 2")
+		if err != nil {
+			return err
+		}
+		id2, _ := res.LastInsertId()
+
+		res, err = q.ExecContext(ctx, "INSERT INTO setup_experience_scripts (team_id, global_or_team_id, name, script_content_id) VALUES (?, ?, ?, ?)", team1.ID, team1.ID, "script1", id1)
+		if err != nil {
+			return err
+		}
+		script1ID, _ = res.LastInsertId()
+
+		res, err = q.ExecContext(ctx, "INSERT INTO setup_experience_scripts (team_id, global_or_team_id, name, script_content_id) VALUES (?, ?, ?, ?)", team2.ID, team2.ID, "script2", id2)
+		if err != nil {
+			return err
+		}
+		script2ID, _ = res.LastInsertId()
+
+		return nil
+	})
+
+	hostTeam1 := "123"
+	hostTeam2 := "456"
+	hostTeam3 := "789"
+
+	anythingEnqueued, err := ds.EnqueueSetupExperienceItems(ctx, hostTeam1, team1.ID)
+	require.NoError(t, err)
+	require.True(t, anythingEnqueued)
+
+	anythingEnqueued, err = ds.EnqueueSetupExperienceItems(ctx, hostTeam2, team2.ID)
+	require.NoError(t, err)
+	require.True(t, anythingEnqueued)
+
+	anythingEnqueued, err = ds.EnqueueSetupExperienceItems(ctx, hostTeam3, team3.ID)
+	require.NoError(t, err)
+	require.False(t, anythingEnqueued)
+
+	seRows := []setupExperienceInsertTestRows{}
+
+	ExecAdhocSQL(t, ds, func(q sqlx.ExtContext) error {
```

Comment:
nit: could we use the list method for this instead?

Reply:
Yes definitely, that was written before the list PR was merged